tests/testthat/test-sma-sim.R

```diff
@@ -1,158 +1,66 @@
+DATAE[, "E"] <- scale(DATAE[, "E"]) * 2
 ## some SMAs require Q also in simulation
 DATAQ <- cbind(P = P, Q = Q)
 
-test_that("cmd simulation is the same in R and C", {
-  set.seed(0)
-  mod0 <- hydromad(DATA, sma = "cmd")
-  for (mod in simulate(mod0, 5)) {
-    Csim <- predict(mod)
-    expect_true(all(Csim >= 0))
-    hydromad.options(pure.R.code = TRUE)
-    pureRsim <- predict(mod)
-    hydromad.options(pure.R.code = FALSE)
-    expect_equal(Csim, pureRsim)
-  }
-})
 
-test_that("cmd power-form simulation is the same in R and C", {
-  set.seed(0)
-  mod0 <- hydromad(DATA, sma = "cmd", shape = 2)
-  for (mod in simulate(mod0, 5)) {
-    Csim <- predict(mod)
-    expect_true(all(Csim >= 0))
-    hydromad.options(pure.R.code = TRUE)
-    pureRsim <- predict(mod)
-    hydromad.options(pure.R.code = FALSE)
-    expect_equal(Csim, pureRsim)
-  }
-})
-
-test_that("cwi simulation is the same in R and C", {
-  set.seed(0)
-  mod0 <- hydromad(DATA, sma = "cwi")
-  for (mod in simulate(mod0, 5)) {
-    Csim <- predict(mod)
-    expect_true(all(Csim >= 0))
-    hydromad.options(pure.R.code = TRUE)
-    pureRsim <- predict(mod)
-    hydromad.options(pure.R.code = FALSE)
-    expect_equal(Csim, pureRsim)
-  }
-})
-
-test_that("bucket simulation is the same in R and C", {
-  set.seed(0)
-  mod0 <- hydromad(DATA, sma = "bucket")
-  for (mod in simulate(mod0, 5)) {
-    Csim <- predict(mod)
-    expect_true(all(Csim >= 0))
-    hydromad.options(pure.R.code = TRUE)
-    pureRsim <- predict(mod)
-    hydromad.options(pure.R.code = FALSE)
-    expect_equal(Csim, pureRsim)
-  }
-})
-
-test_that("gr4j simulation is the same in R and C", {
-  set.seed(0)
-  mod0 <- hydromad(DATA,
-    sma = "gr4j", routing = "gr4jrouting",
-    etmult = 0.1, S_0 = 0.5, R_0 = 0
+with_parameters_test_that(
+  "cmds are the same in R and C",
+  {
+    set.seed(0)
+    n_iter <- 5
+    for (mod in simulate(mod, n_iter)) {
+      Csim <- predict(mod)
+      expect_true(all(Csim >= 0))
+      hydromad.options(pure.R.code = TRUE)
+      pureRsim <- predict(mod)
+      hydromad.options(pure.R.code = FALSE)
+      expect_equal(Csim, pureRsim)
+    }
+  },
+  cases(
+    `sma = "cmd"` = list(mod = hydromad(DATA, sma = "cmd")),
+    `sma = "cmd", shape = 2` = list(mod = hydromad(DATA, sma = "cmd", shape = 2)),
+    `sma = "cwi"` = list(mod = hydromad(DATA, sma = "cwi")),
+    `sma = "bucket"` = list(mod = hydromad(DATA, sma = "bucket")),
+    `sma = "awbm"` = list(mod = hydromad(DATA, sma = "awbm")),
+    `sma = "gr4j"` = list(mod = hydromad(DATA,
+      sma = "gr4j", routing = "gr4jrouting",
+      etmult = 0.1, S_0 = 0.5, R_0 = 0
+    )),
+    `sma = "awbm"` = list(mod = hydromad(DATA, sma = "awbm")),
+    `sma = "simhyd"` = list(mod = hydromad(DATA, sma = "simhyd")),
+    `sma = "sacramento"` = list(mod = hydromad(DATA, sma = "sacramento")),
+    `sma = "snow"` = list(mod = hydromad(DATA, sma = "snow"))
   )
-  for (mod in simulate(mod0, 5)) {
-    Csim <- na.trim(predict(mod))
```

Comment:
Is it possible that this `na.trim`, missing in your patch, is what's reducing code coverage?

Reply:
I think you may be right - I've added code to account for this ... let's see what codecov says.

Reply:
Nope, turned out the problem was the dataset for 'snow'!

@josephguillaume I think this PR is ready to merge.